clojure-ts-mode.el

```diff
@@ -1691,10 +1697,191 @@ Forms between BEG and END are aligned according to
+          (insert " " contents)
+          (when multiline-p
+            (insert "\n")))))))
+
+(defun clojure-ts--unwind-last ()
+  "Unwind a thread last macro once."
+  (let* ((threading-sexp (clojure-ts--threading-sexp-node))
+         (first-child-start (thread-first threading-sexp
+                                          (treesit-node-child 0 t)
+                                          (treesit-node-start)
+                                          (copy-marker))))
+    (save-excursion
+      (goto-char first-child-start)
+      (treesit-beginning-of-thing 'sexp -1)
+      (let ((contents (clojure-ts--delete-and-extract-sexp)))
+        (when (looking-at " *\n")
+          (join-line 'following))
+        (just-one-space)
+        (goto-char first-child-start)
+        (treesit-beginning-of-thing 'sexp -1)
+        (let ((multiline-p (clojure-ts--multiline-sexp-p)))
+          (clojure-ts--ensure-parens-around-function-name)
+          (forward-list)
+          (down-list -1)
+          (when multiline-p
+            (insert "\n"))
+          (insert " " contents))))))
+
+(defun clojure-ts--node-threading-p (node)
+  "Return non-nil if NODE is a threading macro s-expression."
+  (and (or (clojure-ts--list-node-p node)
+           (clojure-ts--anon-fn-node-p node))
+       (let ((first-child (treesit-node-child node 0 t)))
+         (clojure-ts--symbol-matches-p clojure-ts--threading-macro first-child))))
+
+(defun clojure-ts--skip-first-child (parent)
+  "Move point to the beginning of the first child of the PARENT node."
+  (thread-first parent
+                (treesit-node-child 1 t)
+                (treesit-node-start)
+                (goto-char)))
+
+(defun clojure-ts--nothing-more-to-unwind ()
+  "Return TRUE if threading expression at point has only one argument."
+  (let ((threading-sexp (clojure-ts--threading-sexp-node)))
+    (not (save-excursion
+           (clojure-ts--skip-first-child threading-sexp)
+           (treesit-end-of-thing 'sexp 2 'restricted)))))
+
+
+(defun clojure-ts--pop-out-of-threading ()
+  "Raise a sexp up a level to unwind a threading form."
+  (let ((threading-sexp (clojure-ts--threading-sexp-node)))
+    (save-excursion
+      (clojure-ts--skip-first-child threading-sexp)
+      (raise-sexp))))
+
+(defun clojure-ts--fix-sexp-whitespace ()
+  "Fix whitespace after unwinding a threading form."
+  (save-excursion
+    (let ((beg (point)))
+      (treesit-end-of-thing 'sexp)
+      (indent-region beg (point))
+      (delete-trailing-whitespace beg (point)))))
+
+(defun clojure-ts--unwind-sexps-counter ()
+  "Return total number of s-expressions of a threading form at point."
+  (if-let* ((threading-sexp (clojure-ts--threading-sexp-node)))
+      (save-excursion
+        (clojure-ts--skip-first-child threading-sexp)
+        (let ((n 0))
+          (while (treesit-end-of-thing 'sexp 1 'restricted)
+            (setq n (1+ n)))
+          n))
+    (user-error "Nothing to unwind at point")))
```

Comment:
Perhaps we can be a bit more specific and say something like "No threading form to unwind at point"